docs/website/docs/plus/features/transformations/dbt-transformations.md

```diff
@@ -0,0 +1,235 @@
+
+The dbt generator can be used as part of local transformations feature as well as a standalone tool, enabling you to generate dbt models for any dlt pipeline.
+In this context, the dbt generator will be discussed as a standalone feature, though all the information provided is also applicable when using it with local transformations.
+
+The dbt generator works as follows:
+
+- It automatically inspects the pipeline schema and generates a baseline dbt project, complete with staging and marts layers. The generator is able to create staging, dimensional, and fact models.
+
+- Additionally, the dlt dbt generator lets you define relationships between the schema tables, which can be used to automatically create fact tables.
+
+- The resulting project can be executed using the credentials already provided to the pipeline and is capable of processing incoming data incrementally.
+
+## Adding relationship hints for fact tables
+
+To generate fact tables, you will first need to add additional relationship hints to your pipeline. This requires ensuring that each table has a primary key defined, as relationships are based on these keys:
+
+```py
+import dlt
+
+
+@dlt.resource(name="customers", primary_key="id")
+def customers():
+    ...
+
+```
+
+To add relationship hints, use the relationship adapter:
+
+```py
+import dlt
+from dlt_plus.dbt_generator.utils import table_reference_adapter
+
+
+# Example countries table
+@dlt.resource(name="countries", primary_key="id", write_disposition="merge")
+def countries():
+    yield from [
+        {"id": 1, "name": "USA"},
+        {"id": 2, "name": "Germany"},
+    ]
+
+
+# Example companies table
+@dlt.resource(name="companies", primary_key="id", write_disposition="merge")
+def companies():
+    yield from [
+        {"id": 1, "name": "GiggleTech", "country_id": 2},
+        {"id": 2, "name": "HappyHacks", "country_id": 1},
+    ]
+
+
+# Example customers table which references company
+@dlt.resource(name="customers", primary_key="id", write_disposition="merge")
+def customers():
+    yield from [
+        {"id": 1, "name": "Andrea", "company_id": 1},
+        {"id": 2, "name": "Violetta", "company_id": 2},
+        {"id": 3, "name": "Marcin", "company_id": 1},
+    ]
+
+
+# Example orders table which references customer
+@dlt.resource(name="orders", primary_key="id", write_disposition="merge")
+def orders():
+    yield from [
+        {"id": 1, "date": "1-2-2020", "customer_id": 1},
+        {"id": 2, "date": "14-2-2020", "customer_id": 2},
+        {"id": 3, "date": "18-2-2020", "customer_id": 1},
+        {"id": 4, "date": "1-3-2020", "customer_id": 3},
+        {"id": 5, "date": "2-3-2020", "customer_id": 3},
+    ]
+
+# Run your pipeline
+p = dlt.pipeline(pipeline_name="example_shop", destination="duckdb")
+p.run([customers(), companies(), orders(), countries()])
+
+# Define relationships in your schema
+table_reference_adapter(
```

Comment:
This is still incorrect, I had added a note on yesterdays PR. I'm pretty sure you have to defined the references before you run the pipeline, otherwise the relationships are not stored in the schema and thus are not available for the dbt generator.

Reply:
discussed in slack